app.py

```diff
@@ -36,136 +30,195 @@ def send_error_message(self):
 
+####################### PRESCRIPTIONS ########################
+@app.route('/prescriptions/<prescription_id>', methods=['PUT'])
+def put_prescription(prescription_id):
+    try:
+        if not prescriptions.get_prescription(prescription_id):
+            raise IdNotFoundError(prescription_id)
+
+        data_to_modify = json.loads(request.data)
+        data_to_modify.pop('id', None)
+        data_to_modify.pop('created_date', None)
+
+        if not data_to_modify:
+            return json.dumps({'Error': 'ERROR 400 "NO DATA TO MODIFY"'}), 400
+
+        if 'user_id' in data_to_modify:
+            return json.dumps({'Error': 'ERROR 400 "CANNOT MODIFY USER_ID, DELETE PRESCRIPTION AND CREATE A NEW ONE"'}), 400
+
+        prescriptions.modify_prescription(prescription_id, data_to_modify)
+        return {}    
+
+    except IdNotFoundError as e:
+        return json.dumps({'Error': e.send_error_message()}), 404
 
-####################### POST-METHODS #########################
 
+###################################################### POST-METHODS #############################################################
+
+######################### USERS ##############################
 @app.route('/users', methods=['POST'])
-def user_post():
-    print(request.data)
+def post_user():
     user = json.loads(request.data)
     try:
-        if valid_request(user):
-            user['date'] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
-            user['password'] = hash_password(user['password'])
-            
-            sql_response = sql_commands.post_new_user(user)
-                            
-            for value in sql_response[0]:
-                if isinstance(value, int):
-                    user['id'] = value
-            #app.logger.info('USERS RESULT:\n\n{}'.format(user))
-            return json.dumps(user)
+        if validate_user_body(user):
+            user_id = users.post_user(user)
+            return json.dumps({'id': user_id})
 
     except MissingFieldError as e:
         app.logger.debug(e.send_error_message())
         return json.dumps({'Error': e.send_error_message()}), 400
 
-def valid_request(data):
-    for header in ('name', 'email'):
-        if not header in data.keys():
-            app.logger.debug(header)
-            raise MissingFieldError(header)
+
+def validate_user_body(data):
+    for field in ('name', 'email'):
+        if not field in data.keys():
+            app.logger.debug(field)
+            raise MissingFieldError(field)
     return True
 
 
+######################################################
 
-def hash_password(password):
-    return hashlib.md5(password.encode()).hexdigest()
+################## PRESCRIPTIONS #####################
+@app.route('/prescriptions', methods=['POST'])
+def prescription_post():
+    prescription = json.loads(request.data)
+    try:
+        if not users.get_user(prescription['user_id']):
+            raise IdNotFoundError(prescription['user_id'])
```

Comment:
```suggestion
            raise IdNotFoundError(prescription.get('user_id'))
```

Reply:
prescription.get(#receives the prescription_id) who doesn't exist because this is the POST method; so i change this suggestion to that:
        if 'user_id' not in prescription: #prescription is a dict
            raise MissingFieldError('user_id')
        
        if not users.get_user(prescription['user_id']):
            raise IdNotFoundError(prescription['user_id'])